internal/provider/resource_access_generic.go

```diff
@@ -140,27 +155,290 @@ func (a *genericJAASAccessResource) Configure(ctx context.Context, req resource.
+			return
+		}
+	}
+
+	// TODO: Consider updating the state here to reflect the newly added tuples before removing tuples in case the next removal fails.
+	// Would require an intermediate state.
 
+	// Delete removed relations
+	if len(removeTuples) > 0 {
+		err := resource.client.Jaas.DeleteRelations(removeTuples)
+		if err != nil {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to remove access rules, got error: %s", err))
+			return
+		}
+	}
+
+	// Set the desired plan onto the Terraform state after all updates have taken place.
+	resp.Diagnostics.Append(resource.save(ctx, &resp.State, plan, targetTag)...)
+}
+
+func diffModels(plan, state genericJAASAccessModel, diag *diag.Diagnostics) (toAdd, toRemove genericJAASAccessModel) {
+	newUsers := diffSet(plan.Users, state.Users, diag)
+	newGroups := diffSet(plan.Groups, state.Groups, diag)
+	newServiceAccounts := diffSet(plan.ServiceAccounts, state.ServiceAccounts, diag)
+	toAdd.Users = newUsers
+	toAdd.Groups = newGroups
+	toAdd.ServiceAccounts = newServiceAccounts
+	toAdd.Access = plan.Access
+
+	removedUsers := diffSet(state.Users, plan.Users, diag)
+	removedGroups := diffSet(state.Groups, plan.Groups, diag)
+	removedServiceAccounts := diffSet(state.ServiceAccounts, plan.ServiceAccounts, diag)
+	toRemove.Users = removedUsers
+	toRemove.Groups = removedGroups
+	toRemove.ServiceAccounts = removedServiceAccounts
+	toRemove.Access = plan.Access
+
+	return
+}
+
+func diffSet(current, desired basetypes.SetValue, diag *diag.Diagnostics) basetypes.SetValue {
+	var diff []attr.Value
+	for _, source := range current.Elements() {
+		found := false
+		for _, target := range desired.Elements() {
+			if source.Equal(target) {
+				found = true
+			}
+		}
+		if !found {
+			diff = append(diff, source)
+		}
+	}
+	newSet, diags := basetypes.NewSetValue(current.ElementType(context.Background()), diff)
+	diag.Append(diags...)
+	return newSet
 }
 
 // Delete defines how tuples for access control will be updated.
-func (a *genericJAASAccessResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+func (resource *genericJAASAccessResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Check first if the client is configured
+	if resource.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "access model", "delete")
+		return
+	}
+
+	// Read Terraform configuration from the state
+	state, targetTag := resource.info(ctx, req.State, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create the tuples to delete
+	tuples := modelToTuples(ctx, targetTag, state, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Delete the tuples
+	err := resource.client.Jaas.DeleteRelations(tuples)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete access rules, got error: %s", err))
+		return
```

Comment:
nit: no return needed.

Reply:
I'd prefer to keep it in case some code in the future is added after this block.